Dynavity/Dynavity/view/canvas/CanvasView.swift

```diff
@@ -56,47 +22,104 @@ struct CanvasView_Previews: PreviewProvider {
+    func showToolPicker() {
+        toolPicker.setVisible(true, forFirstResponder: annotationCanvasView)
+        toolPicker.addObserver(annotationCanvasView)
+        annotationCanvasView.becomeFirstResponder()
     }
 
-    func handleDragEnd(_ value: DragGesture.Value) {
-        isDragging = false
-        dragOffset = .zero
+    func saveAnnotationToModel() {
+        viewModel.storeAnnotation(annotationCanvasView.drawing)
+    }
 
-        viewPositionFromCentre = CGPoint(
-            x: viewPositionFromCentre.x + value.translation.width,
-            y: viewPositionFromCentre.y + value.translation.height
-        )
+    func didZoom(to scale: CGFloat) {
+        zoomScale = scale
+        cleanAnnotationCanvasView()
+        populateAnnotationCanvasView()
     }
+}
+
+/// `PKCanvasView` is a `UIKit` view. To use it in SwiftUI,
+/// we need to wrap it in a `SwiftUI` view that conforms to UIViewRepresentable.
+extension CanvasView: UIViewRepresentable {
+    func makeUIView(context: Context) -> PKCanvasView {
+        // This simulates an "infinite" canvas
+        let maxContentSize = CGFloat(500_000)
 
-    func scaledOffset(_ scale: CGFloat, initialValue: CGPoint) -> CGPoint {
-        let newX = initialValue.x * scale
-        let newY = initialValue.y * scale
-        return CGPoint(x: newX, y: newY)
+        // For testing purposes on simulator, not necessary otherwise
+        #if targetEnvironment(simulator)
+        annotationCanvasView.drawingPolicy = .anyInput
+        #endif
+
+        annotationCanvasView.delegate = context.coordinator
+        annotationCanvasView.isOpaque = false
+        annotationCanvasView.showsVerticalScrollIndicator = false
+        annotationCanvasView.showsHorizontalScrollIndicator = false
+        annotationCanvasView.contentSize = CGSize(width: maxContentSize, height: maxContentSize)
+        annotationCanvasView.minimumZoomScale = CGFloat(0.1)
+        annotationCanvasView.maximumZoomScale = CGFloat(2.0)
+
+        scrollToInitialContentOffset()
+        showToolPicker()
+        return annotationCanvasView
     }
 
-    func clampedScale(_ scale: CGFloat, initialValue: CGFloat?) -> (scale: CGFloat, didClamp: Bool) {
-        let minScale = CGFloat(0.1)
-        let maxScale = CGFloat(2.0)
-        let raw = scale.magnitude * (initialValue ?? maxScale)
-        let value = raw.clamped(to: minScale...maxScale)
-        let didClamp = raw != value
-        return (value, didClamp)
+    func updateUIView(_ uiView: PKCanvasView, context: Context) {
+        cleanAnnotationCanvasView()
+        populateAnnotationCanvasView()
     }
 
-    func handleScaleChange(_ value: CGFloat) {
-        if initialZoomScale == nil {
-            initialZoomScale = zoomScale
-            initialViewPositionFromCentre = viewPositionFromCentre
+    private func cleanAnnotationCanvasView() {
+        annotationCanvasView.subviews.forEach {
+            if $0.accessibilityLabel == "CanvasElements" {
```

Comment:
Is it possible to remove the view by checking the type instead?

Reply:
So far, I've tried to check by type but the `CanvasElementMapView` is no longer a SwiftUI view, but converted to a `UIView` so that it can be added as a subview to the `AnnotationCanvasView`. So the `UIView` can not be checked if its a `CanvasElementMapView`, hence I added the accessibility label. Any suggestions for this?

Reply:
I see. Then I think it's fine. Thanks!